### DISCUSSION TITLE
Why `query` doesn't have `findOrFail`?

### DISCUSSION THREAD

**Discussion by stereobooster** [adonisjs/core#4028] [Help]
I'm coming from Rails world. Based on my experience I would expect it to work. I guess there can be a limitation which I'm not aware of. Also it can happen that I overlooked something. Now instead of:

```ts
const node = await Node.query().preload('attributes').findOrFail(cntx.params.id)
```

I have to write

```ts
const node = await Node.findOrFail(cntx.params.id)
await node.load('attributes')
```

**Comment by tomgobich** [comment]
Sounds like maybe `firstOrFail` is what you're looking for. https://docs.adonisjs.com/reference/database/query-builder#firstorfail

```ts
const node = await Node.query()
  .preload('attributes')
  .where('id', cntx.params.id)
  .firstOrFail()
```
